### PR TITLE
Azure: Check if container is empty before issuing DeleteContainer

### PR DESCRIPTION
**Pull request by kannappanr**
## Description
Containers in Azure can be deleted even if there are objects within them. Azure gateway should behave like S3 wherever possible. So, it should not delete a bucket which is not empty.

## Motivation and Context
Motivation is to make Azure Gateway behave like S3

## How to test this PR?
Create a bucket using Azure Gatewat and upload an object and issue Delete Bucket S3 API and see the bucket getting deleted.
With this PR, that will not hapen.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

File: cmd/gateway/azure/gateway-azure.go [hunk 1]
```diff
@@ -625,8 +625,17 @@ func (a *azureObjects) ListBuckets(ctx context.Context) (buckets []minio.BucketI
 // DeleteBucket - delete a container on azure, uses Azure equivalent `ContainerURL.Delete`.
 func (a *azureObjects) DeleteBucket(ctx context.Context, bucket string, forceDelete bool) error {
 	containerURL := a.client.NewContainerURL(bucket)
-	_, err := containerURL.Delete(ctx, azblob.ContainerAccessConditions{})
-	return azureToObjectError(err, bucket)
+
```
Comment (vadmeste) [inline cmd/gateway/azure/gateway-azure.go]: DeleteBucket() has forceDelete flag, so we need to return BucketNotEmpty only when forceDelete is set to true.

File: cmd/gateway/azure/gateway-azure.go [hunk 2]
```diff
@@ -625,8 +625,17 @@ func (a *azureObjects) ListBuckets(ctx context.Context) (buckets []minio.BucketI
...
+	resultV1, err := a.ListObjects(ctx, bucket, "", "", "", 1)
+	if err != nil {
+		return azureToObjectError(err, bucket)
+	}
+	if resultV1.Objects == nil || len(resultV1.Objects) == 0 {
```
Comment (vadmeste) [inline cmd/gateway/azure/gateway-azure.go]: `if  len(resultV1.Objects) == 0` is enough

**Review by vadmeste — CHANGES_REQUESTED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#801761a0-6a5d-4107-9cea-63e69a50fd9e) |
<a name="801761a0-6a5d-4107-9cea-63e69a50fd9e"></a>
### 9733-3cf1e62/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev3.minio.io:32086
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 5cb63e47d613:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 7 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 18 seconds
(4/15) Running aws-sdk-ruby tests ... done in 14 seconds
(5/15) Running awscli tests ... done in 2 minutes and 40 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 3 minutes and 56 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 23 seconds
(9/15) Running minio-go tests ... done in 5 minutes and 4 seconds
(10/15) Running minio-java tests ... FAILED in 25 minutes and 30 seconds
error occurred
ErrorResponse(code = BucketNotEmpty, message = The bucket you tried to delete is not empty, bucketName = minio-java-test-3seoq0a, objectName = null, resource = /minio-java-test-3seoq0a, requestId = 16137BABF012FF43, hostId = 10b7a444-ba9d-443d-a1f6-6591cdfd7aca)
request={method=DELETE, url=http://minio-dev3.minio.io:32086/minio-java-test-3seoq0a, headers=Host: minio-dev3.minio.io:32086
Accept-Encoding: identity
User-Agent: MinIO (amd64; amd64) minio-java/dev
Content-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==
x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
x-amz-date: 20200529T111751Z
Authorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200529/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*
}
response={code=409, headers=Accept-Ranges: bytes
Content-Length: 328
Content-Security-Policy: block-all-mixed-content
Content-Type: application/xml
Server: MinIO/DEVELOPMENT.GOGET
Vary: Origin
X-Amz-Request-Id: 16137BABF012FF43
X-Xss-Protection: 1; mode=block
Date: Fri, 29 May 2020 11:17:51 GMT
}

	at io.minio.MinioClient.execute(MinioClient.java:1211)
	at io.minio.MinioClient.execute(MinioClient.java:1236)
	at io.minio.MinioClient.executeDelete(MinioClient.java:1366)
	at io.minio.MinioClient.removeBucket(MinioClient.java:3686)
	at FunctionalTest.teardown(FunctionalTest.java:414)
	at FunctionalTest.runTests(FunctionalTest.java:4122)
	at FunctionalTest.main(FunctionalTest.java:4192)
(10/15) Running minio-js tests ... done in 2 minutes and 38 seconds
(11/15) Running minio-py tests ... done in 11 minutes and 8 seconds
(12/15) Running s3cmd tests ... done in 1 minutes and 25 seconds
(13/15) Running s3select tests ... done in 1 minutes and 2 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

File: cmd/gateway/azure/gateway-azure.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/gateway/azure/gateway-azure.go]: @vadmeste means when forcedelete is set to false

Comment (kannappanr) [reply cmd/gateway/azure/gateway-azure.go]: will do that change.

**Review by harshavardhana — APPROVED** [review]

**Comment by vadmeste** [comment]
By the way, I guess it is okay to ignore incomplete multipart uploads ? I don't see why we need to consider it but just wanted to share with you.

**Comment by harshavardhana** [comment]
> By the way, I guess it is okay to ignore incomplete multipart uploads ? I don't see why we need to consider it but just wanted to share with you.

No we don't care about them @vadmeste S3 delete's bucket with incomplete uploads

**Review by vadmeste — APPROVED** [review]
LGTM & tested

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#95e25923-9667-4d13-93c3-1d6805c9dbe1) |
<a name="95e25923-9667-4d13-93c3-1d6805c9dbe1"></a>
### 9733-31fc7a4/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev3.minio.io:31446
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 4a27c5672035:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 7 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 24 seconds
(4/15) Running aws-sdk-ruby tests ... done in 16 seconds
(5/15) Running awscli tests ... done in 2 minutes and 44 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 3 minutes and 7 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 26 seconds
(9/15) Running minio-go tests ... done in 3 minutes and 50 seconds
(10/15) Running minio-java tests ... FAILED in 25 minutes and 37 seconds
error occurred
ErrorResponse(code = BucketNotEmpty, message = The bucket you tried to delete is not empty, bucketName = minio-java-test-2qm5d6n, objectName = null, resource = /minio-java-test-2qm5d6n, requestId = 16139289DCB34FF9, hostId = 0d46ff36-de31-459b-802b-ca98fb49a986)
request={method=DELETE, url=http://minio-dev3.minio.io:31446/minio-java-test-2qm5d6n, headers=Host: minio-dev3.minio.io:31446
Accept-Encoding: identity
User-Agent: MinIO (amd64; amd64) minio-java/dev
Content-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==
x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
x-amz-date: 20200529T181654Z
Authorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200529/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*
}
response={code=409, headers=Accept-Ranges: bytes
Content-Length: 328
Content-Security-Policy: block-all-mixed-content
Content-Type: application/xml
Server: MinIO/DEVELOPMENT.GOGET
Vary: Origin
X-Amz-Request-Id: 16139289DCB34FF9
X-Xss-Protection: 1; mode=block
Date: Fri, 29 May 2020 18:16:54 GMT
}

	at io.minio.MinioClient.execute(MinioClient.java:1211)
	at io.minio.MinioClient.execute(MinioClient.java:1236)
	at io.minio.MinioClient.executeDelete(MinioClient.java:1366)
	at io.minio.MinioClient.removeBucket(MinioClient.java:3686)
	at FunctionalTest.teardown(FunctionalTest.java:414)
	at FunctionalTest.runTests(FunctionalTest.java:4122)
	at FunctionalTest.main(FunctionalTest.java:4192)
(10/15) Running minio-js tests ... done in 2 minutes and 52 seconds
(11/15) Running minio-py tests ... done in 9 minutes and 50 seconds
(12/15) Running s3cmd tests ... done in 1 minutes and 44 seconds
(13/15) Running s3select tests ... done in 1 minutes and 2 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#b54d23b6-2018-4e44-a315-9faefc3a7319) |
<a name="b54d23b6-2018-4e44-a315-9faefc3a7319"></a>
### 9733-363cf23/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev3.minio.io:31603
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 437af86b0f1f:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 7 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 24 seconds
(4/15) Running aws-sdk-ruby tests ... done in 15 seconds
(5/15) Running awscli tests ... done in 2 minutes and 6 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... FAILED in 49 seconds
{
  "name": "mc",
  "duration": "23244",
  "function": "test_get_object_multipart",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-4041 --quiet --no-color cp myminio/mc-test-bucket-5012/mc-test-object-9314 mc-test-object-9314.downloaded  >>> \n`myminio/mc-test-bucket-5012/mc-test-object-9314` -> `mc-test-object-9314.downloaded`\nmc: <ERROR> Failed to copy `http://minio-dev3.minio.io:31603/mc-test-bucket-5012/mc-test-object-9314`. read tcp 172.17.0.4:41280->72.28.97.52:31603: read: connection reset by peer\nTotal: 0 B, Transferred: 60.50 MiB, Speed: 5.29 MiB/s"
}
(7/15) Running minio-dotnet tests ... done in 1 minutes and 29 seconds
(8/15) Running minio-go tests ... done in 3 minutes and 56 seconds
(9/15) Running minio-java tests ... FAILED in 24 minutes and 1 seconds
error occurred
ErrorResponse(code = BucketNotEmpty, message = The bucket you tried to delete is not empty, bucketName = minio-java-test-38fbcqf, objectName = null, resource = /minio-java-test-38fbcqf, requestId = 16139587F9B82B0A, hostId = c339d01c-005d-46ad-846c-c8f37966093f)
request={method=DELETE, url=http://minio-dev3.minio.io:31603/minio-java-test-38fbcqf, headers=Host: minio-dev3.minio.io:31603
Accept-Encoding: identity
User-Agent: MinIO (amd64; amd64) minio-java/dev
Content-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==
x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
x-amz-date: 20200529T191144Z
Authorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200529/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*
}
response={code=409, headers=Accept-Ranges: bytes
Content-Length: 328
Content-Security-Policy: block-all-mixed-content
Content-Type: application/xml
Server: MinIO/DEVELOPMENT.GOGET
Vary: Origin
X-Amz-Request-Id: 16139587F9B82B0A
X-Xss-Protection: 1; mode=block
Date: Fri, 29 May 2020 19:11:44 GMT
}

	at io.minio.MinioClient.execute(MinioClient.java:1211)
	at io.minio.MinioClient.execute(MinioClient.java:1236)
	at io.minio.MinioClient.executeDelete(MinioClient.java:1366)
	at io.minio.MinioClient.removeBucket(MinioClient.java:3686)
	at FunctionalTest.teardown(FunctionalTest.java:414)
	at FunctionalTest.runTests(FunctionalTest.java:4122)
	at FunctionalTest.main(FunctionalTest.java:4192)
(9/15) Running minio-js tests ... done in 2 minutes and 43 seconds
(10/15) Running minio-py tests ... done in 9 minutes and 39 seconds
(11/15) Running s3cmd tests ... done in 1 minutes and 20 seconds
(12/15) Running s3select tests ... done in 1 minutes and 2 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#96f38f2b-6257-49d9-a647-592f63ebacaf) |
<a name="96f38f2b-6257-49d9-a647-592f63ebacaf"></a>
### 9733-e7a5cde/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:31576
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp e58e1c7547cb:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 7 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 26 seconds
(4/15) Running aws-sdk-ruby tests ... done in 14 seconds
(5/15) Running awscli tests ... done in 2 minutes and 40 seconds
(6/15) Running healthcheck tests ... done in 1 seconds
(7/15) Running mc tests ... done in 3 minutes and 9 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 25 seconds
(9/15) Running minio-go tests ... done in 3 minutes and 49 seconds
(10/15) Running minio-java tests ... FAILED in 24 minutes and 13 seconds
error occurred
ErrorResponse(code = BucketNotEmpty, message = The bucket you tried to delete is not empty, bucketName = minio-java-test-35i2joq, objectName = null, resource = /minio-java-test-35i2joq, requestId = 161398A661FC9EFC, hostId = 02fbe8e7-60e3-440a-a8fe-7334feb411cb)
request={method=DELETE, url=http://minio-c3.minio.io:31576/minio-java-test-35i2joq, headers=Host: minio-c3.minio.io:31576
Accept-Encoding: identity
User-Agent: MinIO (amd64; amd64) minio-java/dev
Content-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==
x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
x-amz-date: 20200529T200853Z
Authorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200529/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*
}
response={code=409, headers=Accept-Ranges: bytes
Content-Length: 328
Content-Security-Policy: block-all-mixed-content
Content-Type: application/xml
Server: MinIO/DEVELOPMENT.GOGET
Vary: Origin
X-Amz-Request-Id: 161398A661FC9EFC
X-Xss-Protection: 1; mode=block
Date: Fri, 29 May 2020 20:08:53 GMT
}

	at io.minio.MinioClient.execute(MinioClient.java:1211)
	at io.minio.MinioClient.execute(MinioClient.java:1236)
	at io.minio.MinioClient.executeDelete(MinioClient.java:1366)
	at io.minio.MinioClient.removeBucket(MinioClient.java:3686)
	at FunctionalTest.teardown(FunctionalTest.java:414)
	at FunctionalTest.runTests(FunctionalTest.java:4122)
	at FunctionalTest.main(FunctionalTest.java:4192)
(10/15) Running minio-js tests ... done in 2 minutes and 50 seconds
(11/15) Running minio-py tests ... done in 10 minutes and 10 seconds
(12/15) Running s3cmd tests ... done in 1 minutes and 37 seconds
(13/15) Running s3select tests ... done in 1 minutes and 2 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#127f7f8c-8893-4069-acdc-f612904812f2) |
| mint-gateway-azure.sh | :x: [more...](#22808f05-0f37-49dd-a4e2-ac8d97eaa470) |
<a name="127f7f8c-8893-4069-acdc-f612904812f2"></a>
### 9733-af95ae8/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:30397
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 245988277dcd:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 1 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 42 seconds
(4/15) Running aws-sdk-ruby tests ... done in 1 seconds
(5/15) Running awscli tests ... done in 2 minutes and 13 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 33 seconds
(8/15) Running minio-dotnet tests ... done in 31 seconds
(9/15) Running minio-go tests ... done in 47 seconds
(10/15) Running minio-java tests ... done in 54 seconds
(11/15) Running minio-js tests ... FAILED in 18 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 20,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}
(11/15) Running minio-py tests ... done in 2 minutes and 26 seconds
(12/15) Running s3cmd tests ... done in 20 seconds
(13/15) Running s3select tests ... done in 2 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="22808f05-0f37-49dd-a4e2-ac8d97eaa470"></a>
### 9733-af95ae8/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:32003
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 6d4976f38ad7:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 27 seconds
(4/15) Running aws-sdk-ruby tests ... done in 15 seconds
(5/15) Running awscli tests ... done in 2 minutes and 53 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 3 minutes and 7 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 24 seconds
(9/15) Running minio-go tests ... done in 3 minutes and 44 seconds
(10/15) Running minio-java tests ... FAILED in 23 minutes and 45 seconds
[minio-java-test-3ed6fg9]: threaded put object
java.io.EOFException
	at okio.RealBufferedSink.write(RealBufferedSink.java:115)
	at io.minio.HttpRequestBody.writeTo(HttpRequestBody.java:76)
	at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:83)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:45)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:94)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:125)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:147)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:121)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:264)
	at okhttp3.RealCall.execute(RealCall.java:93)
	at io.minio.MinioClient.execute(MinioClient.java:1097)
	at io.minio.MinioClient.execute(MinioClient.java:1236)
	at io.minio.MinioClient.executePut(MinioClient.java:1410)
	at io.minio.MinioClient.executePut(MinioClient.java:1424)
	at io.minio.MinioClient.uploadPart(MinioClient.java:5492)
	at io.minio.MinioClient.putObject(MinioClient.java:3740)
	at io.minio.MinioClient.putObject(MinioClient.java:3822)
	at PutObjectRunnable.run(PutObjectRunnable.java:42)
	at java.base/java.lang.Thread.run(Thread.java:834)
error occurred
ErrorResponse(code = BucketNotEmpty, message = The bucket you tried to delete is not empty, bucketName = minio-java-test-14pnt7u, objectName = null, resource = /minio-java-test-14pnt7u, requestId = 16139BCE933A3DFC, hostId = 1f7d38d3-4b74-46d0-9950-b7b264abbbcd)
request={method=DELETE, url=http://minio-dev7.minio.io:32003/minio-java-test-14pnt7u, headers=Host: minio-dev7.minio.io:32003
Accept-Encoding: identity
User-Agent: MinIO (amd64; amd64) minio-java/dev
Content-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==
x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855
x-amz-date: 20200529T210644Z
Authorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20200529/us-east-1/s3/aws4_request, SignedHeaders=accept-encoding;content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*
}
response={code=409, headers=Accept-Ranges: bytes
Content-Length: 328
Content-Security-Policy: block-all-mixed-content
Content-Type: application/xml
Server: MinIO/DEVELOPMENT.GOGET
Vary: Origin
X-Amz-Request-Id: 16139BCE933A3DFC
X-Xss-Protection: 1; mode=block
Date: Fri, 29 May 2020 21:06:45 GMT
}

	at io.minio.MinioClient.execute(MinioClient.java:1211)
	at io.minio.MinioClient.execute(MinioClient.java:1236)
	at io.minio.MinioClient.executeDelete(MinioClient.java:1366)
	at io.minio.MinioClient.removeBucket(MinioClient.java:3686)
	at FunctionalTest.teardown(FunctionalTest.java:414)
	at FunctionalTest.runTests(FunctionalTest.java:4122)
	at FunctionalTest.main(FunctionalTest.java:4192)
(10/15) Running minio-js tests ... done in 2 minutes and 39 seconds
(11/15) Running minio-py tests ... done in 9 minutes and 41 seconds
(12/15) Running s3cmd tests ... done in 2 minutes and 24 seconds
(13/15) Running s3select tests ... done in 1 minutes and 1 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Pull request merged**